articles/an-introduction-to-graph-neural-network/index.md

```diff
@@ -0,0 +1,188 @@
+
+### What is Graph Neural Network (GNN)?
+
+Graph Neural Networks is a technique in deep learning which extends existing neural networks for processing data on graphs. 
+
+GNNs motivation stems from three areas of study:
+
+* Convolution Neural Networks (CNN)
+These are deep neural networks used to analyze image data. It is used to solve image processing tasks.
+
+* Recurrent Neural Networks (RNN)
+RNN is a type of neural network that passes information internally from one time step to the next to process a sequence. It's commonly used to solve Natural Language Processing (NLP) and speech recognition tasks.  
+
+* Auto-Encoders (AE)
+This is a type of Artificial Neural Network (ANN). It consists of an encoder, code, and a decoder. It encodes the representations of a set of data. It then decodes the reduced encoded data generating new data similar to that of its original input. 
+
+Based on these three motivations, GNNs can collectively combine information from the graph structure.
+
+Common applications of GNN are in node classification, link prediction, and graph classification. 
+
+In **node classification**, every node in the network is assigned a label. The network can classify new nodes introduced without the ground truth.
+In **link prediction**, the goal is to  predicts the likelihood of two nodes being inter-linked.
+In **graph classification**, the idea is to classify graphs into different classes. This is related to image classification but the target changes into classifying graphs rather than images.
+
+### Graph Learning Python Libraries
+
+Don't know where to start?
+
+Well. Several freely available tools will let you start working on your project using GNN. 
+
+The top three libraries include:
+
+1. PyTorch Geometric
+
+It has the latest types of Graph Networks already built in for you. This graph networks are available as single line functions that are ready to be called in the PyTorch library. 
+
+Read more about it [here](/https://pytorch-geometric.readthedocs.io/en/latest/) 
+
+2. Graph Nets
+
+It is a python library created by DeepMind Technologies. It helps build graph networks in platforms such as TensorFlow and Sonnet.
+This library provides a lot of documentation and ready-made collab notebooks to showcase how to use their graph network library.
+
+However, it is important to note that their code isn't beginner-friendly when compared to the other two libraries. It is not advisable to start with this library.
+
+Learn more about it [here](/https://github.com/deepmind/graph_nets/)
+
+3. Deep Graph Library (DGL)
+
+DGL was created by the Distributed Machine Learning community on GitHub. This platform has readable code, maintained, and cross-platform.
+
+DGL is the top pick for beginners.
+
+Learn more about it on their official [website](/https://www.dgl.ai/)
+
+### Models of Graph Neural Networks
```

Comment:
Add images for each type of GNN. The reader does not know how a GNN looks. Now we are introducing additional complexity with recurrence, convolutions, encoding, and decoding operations. Visualization helps the reader better understand. Also, mention the applications of each of these. 

The main purpose of introductory articles is to introduce concepts in depth. At the end of the article, the reader should know what they don't know. The additional resources should guide them to those concepts.